mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/MantisStorageProvider.java

```diff
@@ -183,4 +183,12 @@ List<MantisWorkerMetadataWritable> getArchivedWorkers(String jobid)
     void deleteExpiredDisableTaskExecutorRequest(DisableTaskExecutorsRequest request) throws IOException;
 
     List<DisableTaskExecutorsRequest> loadAllDisableTaskExecutorsRequests(ClusterID clusterID) throws IOException;
+
+    void storeNewJobArtifact(JobArtifact jobArtifact) throws JobArtifactAlreadyExistsException, IOException;
+
+    void initJobArtifacts();
```

Comment:
nit: It appears that this method only has side effects instead of returning the current snapshot. However, currently, MantisStorageProvider is not implemented as an actor or as a service. Instead, it's purely used as a data access layer. Given that, wondering if we should return a List<JobArtifact> here.